applications/plugins/SofaPython/Binding_SofaModule.cpp

```diff
@@ -582,6 +586,242 @@ static PyObject * Sofa_getAvailableComponents(PyObject * /*self*/, PyObject * ar
+ * Param  : PyObject*, self - Object of the python script
+ * Param  : PyObject*, args - given arguments to apply to the method
+ * Return : NULL
+ */
+static PyObject * Sofa_timerSetEnabled(PyObject* /*self*/, PyObject *args)
+{
+    char* id;
+    PyObject* val;
+
+    if(!PyArg_ParseTuple(args, "sO", &id, &val))
+    {
+        PyErr_BadArgument();
+        Py_RETURN_NONE;
+    }
+
+    AdvancedTimer::setEnabled(id, PyObject_IsTrue(val));  // Method call
+    Py_RETURN_NONE;
+}
+
+
+/**
+ * Method : Sofa_getInterval
+ * Desc   : Wrapper for python usage.
+ * Param  : PyObject*, self - Object of the python script
+ * Param  : PyObject*, args - given arguments to apply to the method
+ * Return : NULL
+ */
+static PyObject * Sofa_timerGetInterval(PyObject* /*self*/, PyObject *args)
+{
+    char* id;
+    int answer = 0;
+
+    if(!PyArg_ParseTuple(args, "s", &id))
+    {
+        PyErr_BadArgument();
+        return NULL;
+    }
+
+    answer = AdvancedTimer::getInterval(id);  // Method call
+
+    return PyInt_FromLong(static_cast<long int>(answer));
+}
+
+
+/**
+ * Method : Sofa_setInterval
+ * Desc   : Wrapper for python usage.
+ * Param  : PyObject*, self - Object of the python script
+ * Param  : PyObject*, args - given arguments to apply to the method
+ * Return : NULL
+ */
+static PyObject * Sofa_timerSetInterval(PyObject* /*self*/, PyObject *args)
+{
+    char* id;
+    int newValue = 0;
+
+    if(!PyArg_ParseTuple(args, "si", &id, &newValue))
+    {
+        return NULL;
+    }
+
+    AdvancedTimer::setInterval(id, newValue);  // Method call
+
+
+    Py_RETURN_NONE;
+}
+
+
+/**
+ * Method : Sofa_begin
+ * Desc   : Wrapper for python usage.
+ * Param  : PyObject*, self - Object of the python script
+ * Param  : PyObject*, args - given arguments to apply to the method
+ * Return : NULL
+ */
+static PyObject * Sofa_timerBegin(PyObject* /*self*/, PyObject *args)
+{
+    char* id;
+
+    if(!PyArg_ParseTuple(args, "s", &id))
+    {
+        PyErr_BadArgument();
+        return NULL;
+    }
+
+    AdvancedTimer::begin(id);  // Method call
+
+    Py_RETURN_NONE;
+}
+
+
+/**
+ * Method : Sofa_timerStepBegin
+ * Desc   : Wrapper for python usage.
+ * Param  : PyObject*, args - given arguments to apply to the method
+ * Return : NULL
+ */
+static PyObject * Sofa_timerStepBegin(PyObject*, PyObject* args)
+{
+    char* id;
+
+    if(!PyArg_ParseTuple(args, "s", &id))
+    {
+        PyErr_BadArgument();
+        return NULL;
+    }
+
+    AdvancedTimer::stepBegin(id);  // Method call
+
+    Py_RETURN_NONE;
+}
+
+
+/**
+ * Method : Sofa_timerStepEnd
+ * Desc   : Wrapper for python usage.
+ * Param  : PyObject*, args - given arguments to apply to the method
+ * Return : NULL
+ */
+static PyObject * Sofa_timerStepEnd(PyObject*, PyObject* args)
+{
+    char* id;
+
+    if(!PyArg_ParseTuple(args, "s", &id))
+    {
+        PyErr_BadArgument();
```

Comment:
Setting the PyErr_BadArgument() is not needed as PyArg_ParseTuple already set the correct error message. 
(this comment should apply to each PyErr_BadArgument.